jinahub/encoders/image/CLIPImageEncoder/clip_image.py

```diff
@@ -81,35 +80,34 @@ def encode(self, docs: Optional[DocumentArray], parameters: dict, **kwargs):
             The accepted keys are ``traversal_paths`` and ``batch_size`` - in their
             absence their corresponding default values are used.
         """
-        if docs:
-            document_batches_generator = get_docs_batch_generator(
-                docs,
-                traversal_path=parameters.get(
-                    "traversal_paths", self.default_traversal_paths
-                ),
-                batch_size=parameters.get("batch_size", self.default_batch_size),
-                needs_attr="blob",
-            )
+        if docs is None:
+            return
 
-            with torch.no_grad():
-                for batch_docs in document_batches_generator:
-                    blob_batch = [d.blob for d in batch_docs]
-                    if self.use_default_preprocessing:
-                        tensor = self._generate_input_features(blob_batch.copy())
-                    else:
-                        tensor = {
-                            "pixel_values": torch.tensor(
-                                blob_batch.copy(),
-                                dtype=torch.float32,
-                                device=self.device,
-                            )
-                        }
+        traversal_paths = parameters.get("traversal_paths", self.traversal_paths)
+        batch_size = parameters.get("batch_size", self.batch_size)
+        document_batches_generator = docs.batch(
+            traversal_paths=traversal_paths,
+            batch_size=batch_size,
+            require_attr="blob",
+        )
 
-                    embeddings = self.model.get_image_features(**tensor)
-                    embeddings = embeddings.cpu().numpy()
+        with torch.inference_mode():
+            for batch_docs in document_batches_generator:
+                blob_batch = batch_docs.blobs
+                if self.use_default_preprocessing:
+                    tensor = self._generate_input_features(blob_batch.copy())
```

Comment:
why is the copy needed?

Reply:
I thought it might otherwise modify the blob in place, not sure

Reply:
I think it should not. the blob_batch is a numpy array with no access to the protobuf underlying

Reply:
Also it is weird to change the input to generate something.

Reply:
Okay, that makes sense